### PR TITLE
lighting js MainMenu commponent

### PR DESCRIPTION
**Pull request by kathiravanltts**
Refapp to refapp2 (lighting js) application pull request.
this pull request includes the below features
1. Basic list feature added. This list component will provide dynamic list and get data from model(json file).
2. Main menu component added. Main Menu having three list-item such as “App”,”Movie”,”Setting”.
3. Dummy screen added for list-item(“App”,”Movie”) action.
3. This pull request changes tested on browser-Chrome.

### CODE REVIEW

**Comment by dwrobel** [comment]
- use a single PR for further work - until it got merged _(I closed https://github.com/LibertyGlobal/refapp/pull/12)_,
- [ ] do not add any new fonts _(pixel.ttf)_ as they usually have non-acceptable license _(rely only on those fonts which are available by default on the platform)_,
- [ ] remove the `static` directory _(leave origin config.json and config.ssm.json files untouched as discussed on https://rdkm.slack.com/archives/GFAFFDS9H/p1585225666030700)_,
  - do not mix fonts, json, and jpg files in one single directory,
  - [ ] do not add any content/images to which we don't have rights to re-distribute _(static/img.jpg)_
    - preferably reuse existing images from src/img,
     - if you need to add a new image add it to the existing src/img directory,
 - [ ] `npm install -g WebPlatformForEmbedded/Lightning-CLI` - do not enforce user to perform system-wide installation as it might not be possible _(due to access restriction)_.

_Mark individual tasks as done when fixed._

**Comment by piotr-serafin-epam** [comment]
- [ ] add dist to .gitignore

**Comment by kathiravanltts** [comment]
> * use a single PR for further work - until it got merged _(I closed #12)_,
> * [x]  do not add any new fonts _(pixel.ttf)_ as they usually have non-acceptable license _(rely only on those fonts which are available by default on the platform)_,


> * [x]  remove the `static` directory _(leave origin config.json and config.ssm.json files untouched as discussed on https://rdkm.slack.com/archives/GFAFFDS9H/p1585225666030700)_,
>   
>   * do not mix fonts, json, and jpg files in one single directory,
>   * [x]  do not add any content/images to which we don't have rights to re-distribute _(static/img.jpg)_
>     
>     * preferably reuse existing images from src/img,
>     * if you need to add a new image add it to the existing src/img directory,
> * [x]  `npm install -g WebPlatformForEmbedded/Lightning-CLI` - do not enforce user to perform system-wide installation as it might not be possible _(due to access restriction)_.
> 
> _Mark individual tasks as done when fixed._

**Comment by kathiravanltts** [comment]
1. Done, pixel.ttf fonts are removed.  Platform font took from “/usr/share/fonts

2. Done,static folder removed from github.so added buildsupport.sh script files. The  required “runtime shared file” copying from src folder to disit folder.
Folders are separated to store json,img,font..

3.Done, Lightning-CLI commands is required to build our application.
reference: https://webplatformforembedded.github.io/Lightning/docs/gettingStarted/setup-lightning-sdk

4.Done, disit folder name added into .gitgnore list.

File: dist/settings.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "appSettings": {
+    "stage": {
+      "clearColor": "0x00000000",
+      "useImageWorker": true
```
Comment (piotr-serafin-epam) [inline dist/settings.json]: for now use false

File: dist/settings.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    },
+    "debug": false
+  },
+  "platformSettings": {
+    "path": "./static",
+    "log": true,
+    "showVersion": true
```
Comment (piotr-serafin-epam) [inline dist/settings.json]: Add "esEnv": "es5"

Comment (kathiravanltts) [reply dist/settings.json]: added  into ./setting.json

File: dist/settings.json (hunk 1 above)
Comment (kathiravanltts) [reply dist/settings.json]: added  into ./setting.json

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import { Lightning, Utils } from 'wpe-lightning-sdk'
+import Menu from './mainMenu/menu.js'
+import OnDemand from './OnDemand/ondemand.js'
+
+export default class App extends Lightning.Component {
+  static getFonts() {
+    return [{ family: 'pixel', url: Utils.asset('../fonts/DejaVuSerif.ttf'), descriptor: {} }]
```
Comment (dwrobel) [inline src/App.js:7]: As agreed please remove the DejaVuSerif.ttf font binary blob.

File: metadata.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "My Awesome App",
```
Comment (piotr-serafin-epam) [inline metadata.json:2]: Please use proper name and identifier eg:
"name": "Refapp2"
"identifier": "com.rdkcentral.app.refapp2"

File: src/App.js (hunk 3 above)
Comment (kathiravanltts) [reply src/App.js]: font removed from getFonts method.

File: metadata.json (hunk 4 above)
Comment (kathiravanltts) [reply metadata.json]: metadata.json  update done.

**Comment by kathiravanltts** [comment]
> * use a single PR for further work - until it got merged _(I closed #12)_,
> * [x]  do not add any new fonts _(pixel.ttf, DejaVuSerif.ttf)_ as they usually have non-acceptable license _(rely only on those fonts which are available by default on the platform)_,
> * [x]  remove the `static` directory _(leave origin config.json and config.ssm.json files untouched as discussed on https://rdkm.slack.com/archives/GFAFFDS9H/p1585225666030700)_,
>   
>   * do not mix fonts, json, and jpg files in one single directory,
>   * [x]  do not add any content/images to which we don't have rights to re-distribute _(static/img.jpg)_
>     
>     * preferably reuse existing images from src/img,
>     * if you need to add a new image add it to the existing src/img directory,
> * [x]  `npm install -g WebPlatformForEmbedded/Lightning-CLI` - do not enforce user to perform system-wide installation as it might not be possible _(due to access restriction)_.
> 
> _Mark individual tasks as done when fixed._

**Pull request merged**